graphql-ballerina/modules/parser/parser.bal

```diff
@@ -164,13 +177,32 @@ public class Parser {
     }
 
     isolated function addFragmentToNode(ParentNode parentNode) returns ([string, Location]|Error) {
-        Token token = check self.readNextNonSeparatorToken(); // Consume Ellipsis token
-        token = check self.readNextNonSeparatorToken();
+        Token token = check self.readNextNonSeparatorToken();
         string fragmentName = check getIdentifierTokenvalue(token);
         parentNode.addFragment(fragmentName);
         return [fragmentName, token.location];
     }
 
+    isolated function addInlineFragmentToNode(ParentNode parentNode) returns ([string, Location]|Error) {
+        Token token = check self.readNextNonSeparatorToken();//Consume on keyword
+        Location location = token.location.clone();
+        token = check self.readNextNonSeparatorToken();
+        string onType = check getIdentifierTokenvalue(token);
+        if (onType == ON) {
+            return getUnexpectedTokenError(token);
```

Comment:
This should not be a parsing errror, but a validation error.
Named fragments cannot have `on` as the fragment name, we don't have that issue with the inline fragments. So, we should add the type name as it is, and when validating the fragment, it should throw an error `Unknown type \"on\".`

We can remove this if condition completely. (Let's add a test for this scenario as well to check whether the validation phase throws this error).